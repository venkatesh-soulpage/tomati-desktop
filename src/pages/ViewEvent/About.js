import React, { useState, useEffect } from "react";
import { Form, Button } from "react-bootstrap";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { getEvent } from "_actions/event";
import * as Action from "_actions";
import "react-datetime/css/react-datetime.css";
import CustomModal from "components/CustomModal";
import Success from "assets/img/Success.svg";

const About = (props) => {
  useEffect(() => {
    getEvent(props.location.state);
  }, []);

  const [show, setShow] = useState(false);
  const [message, setMessage] = useState("");

  const { event } = props.event;
  const [values, setValues] = useState({
    name: event && event.name,
    description: event && event.description,
    address: event && event.address,
    phone_number: event && event.phone_number,
  });

  const handleChange = (e) => {
    const val = e.target.value;
    setValues({
      ...values,
      [e.target.name]: val,
    });
  };

  const handleUpdate = async () => {
    console.log(values);
    await props.dispatch(Action.updateEvent(props.location.state, values));
    if (props.event.message) {
      setMessage(props.event.message);
      setShow(true);
    }
  };
  let inputProps = {
    placeholder: "Start Time",
  };
  let inputProps2 = {
    placeholder: "End Time",
  };
  return (
    <div className="card bg-white border p-5 mt-2">
      <div className="d-flex align-items-center">
        <div className="mr-auto">
          <label>Event Name</label>
          <Form.Group>
            <Form.Control
              type="text"
              value={values.name}
              name="name"
              required
              onChange={handleChange}
            />
          </Form.Group>
          <label>Phone Nmber</label>
          <Form.Group>
            <Form.Control
              type="text"
              value={values.phone_number}
              name="phone_number"
              required
              onChange={handleChange}
            />
          </Form.Group>
          <label>Event Description</label>

          <Form.Group>
            <Form.Control
              type="text"
              as="textarea"
              rows={3}
              placeholder="FullName"
              name="description"
              value={values.description}
              required
              onChange={handleChange}
            />
          </Form.Group>
          <label>Event Address</label>
          <Form.Group>
            <Form.Control
              type="text"
              as="textarea"
              rows={3}
              placeholder="FullName"
              name="address"
              value={values.address}
              required
              onChange={handleChange}
            />
          </Form.Group>
          {/* <label>Start Time</label>
          <Form.Group>
            <Datetime
              inputProps={inputProps}
              value={moment(values.start_time).format("MMMM Do YYYY, h:mm a")}
              onChange={(e) =>
                setValues({ ...values, start_time: moment(e).format() })
              }
            />
          </Form.Group>
          <label>End Time</label>

          <Form.Group>
            <Datetime
              inputProps={inputProps2}
              //   onChange={handleChange("end_time")}
              value={moment(values.start_time).format("MMMM Do YYYY, h:mm a")}
              onChange={(e) =>
                setValues({ ...values, end_time: moment(e).format() })
              }
            />
          </Form.Group> */}
          <button
            className="btn btn-danger rounded-pill"
            onClick={handleUpdate}
          >
            Save
          </button>
        </div>
      </div>
      <CustomModal
        show={show}
        message={message}
        onHide={() => setShow(false)}
        statusicon={Success}
        button={
          <Button
            className="btn btn-primary mt-3 rounded-pill px-4 py-2"
            onClick={() => setShow(false)}
          >
            Close
          </Button>
        }
      />
    </div>
  );
};

function mapStateToProps(state) {
  return { event: state.event };
}

export default withRouter(connect(mapStateToProps)(About));
