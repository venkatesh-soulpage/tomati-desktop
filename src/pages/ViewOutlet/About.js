import React, { useState, useEffect } from "react";
import { Form } from "react-bootstrap";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { getOutlet, updateOutlet } from "_actions/outlet";
import Success from "assets/img/Success.svg";
import CustomModal from "components/CustomModal";

const About = (props) => {
  useEffect(() => {
    getOutlet(props.location.state);
  }, []);
  const [show, setShow] = useState(false);
  const [message, setMessage] = useState("");

  const { outlet } = props.outlet;
  const [values, setValues] = useState({
    name: outlet && outlet.name,
    description: outlet && outlet.description,
    address: outlet && outlet.address,
    phone_number: outlet && outlet.phone_number,
  });

  const handleChange = (e) => {
    const val = e.target.value;
    setValues({
      ...values,
      [e.target.name]: val,
    });
  };

  const handleUpdate = () => {
    console.log(values);
    props.dispatch(updateOutlet(props.location.state, values)).then((res) => {
      setMessage(res);
      setShow(true);
    });
  };
  return (
    <div className="card bg-white border p-5 mt-2">
      <div className="d-flex align-items-center">
        <div className="mr-auto">
          <label>Outlet Name</label>
          <Form.Group>
            <Form.Control
              type="text"
              value={values.name}
              name="name"
              required
              onChange={handleChange}
            />
          </Form.Group>
          <label>Phone Number</label>
          <Form.Group>
            <Form.Control
              type="text"
              value={values.phone_number}
              name="phone_number"
              required
              onChange={handleChange}
            />
          </Form.Group>
          <label>Outlet Description</label>

          <Form.Group>
            <Form.Control
              type="text"
              as="textarea"
              rows={3}
              name="description"
              value={values.description}
              required
              onChange={handleChange}
            />
          </Form.Group>
          <label>Outlet Address</label>

          <Form.Group>
            <Form.Control
              type="text"
              as="textarea"
              rows={3}
              name="address"
              value={values.address}
              required
              onChange={handleChange}
            />
          </Form.Group>
          <button
            className="btn btn-danger rounded-pill"
            onClick={handleUpdate}
          >
            Save
          </button>
        </div>
      </div>
      <CustomModal
        show={show}
        message={message}
        onHide={() => setShow(false)}
        statusicon={Success}
      />
    </div>
  );
};

function mapStateToProps(state) {
  return { outlet: state.outlet };
}

export default withRouter(connect(mapStateToProps)(About));
