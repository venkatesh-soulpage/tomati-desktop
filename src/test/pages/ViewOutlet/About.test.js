import React from "react";
import { render } from "@testing-library/react";
import About from "pages/ViewOutlet/About";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("About page", () => {
  it("matches snapshot", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <About />
        </Router>
      </Provider>,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
