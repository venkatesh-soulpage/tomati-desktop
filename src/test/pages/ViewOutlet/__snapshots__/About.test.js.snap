// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card bg-white border p-5 mt-2"
  >
    <div
      class="d-flex align-items-center"
    >
      <div
        class="mr-auto"
      >
        <label>
          Menu Name
        </label>
        <div
          class="form-group"
        >
          <input
            class="form-control"
            name="name"
            required=""
            type="text"
            value=""
          />
        </div>
        <label>
          Phone Number
        </label>
        <div
          class="form-group"
        >
          <input
            class="form-control"
            name="phone_number"
            required=""
            type="text"
            value=""
          />
        </div>
        <label>
          Description
        </label>
        <div
          class="form-group"
        >
          <textarea
            class="form-control"
            maxlength="144"
            name="description"
            required=""
            rows="3"
            type="text"
          />
        </div>
        <label>
          Address
        </label>
        <div
          class="form-group"
        >
          <textarea
            class="form-control"
            name="address"
            required=""
            rows="3"
            type="text"
          />
        </div>
        <button
          class="btn btn-danger rounded-pill"
        >
          Save
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
